src/diffusers/models/attention.py

```diff
@@ -55,12 +55,18 @@ def __init__(
         norm_num_groups: int = 32,
         rescale_output_factor: float = 1.0,
         eps: float = 1e-5,
+        use_spatial_norm: bool = False,
```

Comment:
This class is deprecated and should not be used anymore cc @williamberman big time we remove it ;-) 

Let's please try to solve this in the other attention class

Reply:
@yiyixuxu Can you help to instead make this work  with the `Attention` class in `attention_processor`?

Reply:
@patrickvonplaten 

ok. Just to confirm - we should update these blocks to use the `Attention` class instead? 
`UNetMidBlock2D`, `AttnUpDecoderBlock2D` and `AttnDownEncoderBlock2D`

Reply:
We will need to use `AttnAddedKVProcessor` since to replicate the attention used in Kandinksy, we need residual connections and spatial-norm on the hiddens states before passing for q-k-v calculations, but it has this concatenation thing going on [here](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py#L593), which needs to be tackled.

Reply:
@ayushtues ohh I don't think we should use `AttnAddedKVProcessor` since we don't have these additional projection layers for k and v as the processor name indicated 😁

for attention processors, I think we can either
1. create a new attention processor, e.g. `MOVQAttnProcessor` 
2. or make it work with `AttnProcessor`  (https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py#L381), we need to:
    - add an argument to allow skip-connection
    - add the norm layer, maybe before this line, we can add  https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py#L393
    ```python
      if self.group_norm:
          hidden_states = self.group_norm(hidden_states)
    ```

we also need slightly refactor the`Attention` class here (https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py#L92) so the `group_norm` layer can be `spatial_norm`

I would wait for @patrickvonplaten to clarify how we should approach this. It's a little bit tricky IMO because it's overlapping a larger effort here to completely replace the deprecated `AttentionBlock` class

Reply:
To me, this sounds like a better approach:

> create a new attention processor, e.g. MOVQAttnProcessor

(with the new `Attention` class) 

I think this helps to eliminate any overlap between this PR and the larger effort for refactoring the attention blocks.

Reply:
@ayushtues great! let me know once you need another review:) we need to add tests too once everything looks good

Reply:
I instead added a `MOVQAttention` in `unet_2d_blocks` here https://github.com/ayushtues/diffusers/blob/kandinsky_decoder/src/diffusers/models/unet_2d_blocks.py#L1957, which uses the basic AttentionProcessor, and handles all the other processing needed in it, so that we need to make no changes in either `attention.py` or `attention_processor.py`, something similar seemed to be done here in the past: https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/t5_film_transformer.py#L185

Reply:
Let me know how this looks, and maybe we can move this block elsewhere if needed